src/main/java/hudson/plugins/active_directory/ActiveDirectoryDomain.java

```diff
@@ -197,6 +204,13 @@ public TlsConfiguration getTlsConfiguration() {
         return tlsConfiguration;
     }
 
+    protected Object readResolve() {
+        if (isFipsNonCompliant(tlsConfiguration != null && TlsConfiguration.TRUST_ALL_CERTIFICATES.equals(tlsConfiguration))) {
```

Comment:
```suggestion
        if (isFipsNonCompliant(tlsConfiguration != null || TlsConfiguration.TRUST_ALL_CERTIFICATES.equals(tlsConfiguration))) {
```
Probably still better to use the existing method for consistency.